### PR TITLE
chore: swb dynamic authorization infrastructure

### PR DESCRIPTION
**Pull request by insignias**
Issue #, if available:

Description of changes:
1. SWB Dynamic Authorization Infrastructure update

Checklist:

* [ ] Have you successfully deployed to an AWS account with your changes?
* [ ] Have you written new tests for your core changes, as applicable?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by DaveMadsen — DISMISSED** [review]

**Review by aws-tyler — CHANGES_REQUESTED** [review]
To prevent merge conflicts from happening with my branch, lets remove the `swb-app` code. We're not allowing any new references to external projects in `swb-app` so we can make sure we're not tightly coupling with anything that could change without our knowledge. 

The only other thing is I noticed there are no tests. Is there anything we can do to make sure the infrastructure doesn't change unexpectedly?

Once this is updated, I can pull this into my branch ASAP and we can get the service merged into develop 🎉🎉🎉

File: solutions/swb-app/src/apiRouteConfig.ts [hunk 1]
```diff
@@ -4,6 +4,7 @@
  */
 
 import { CostCenterService, HostingAccountService, ProjectService } from '@aws/workbench-core-accounts';
+import { DynamicAuthorizationService } from '@aws/workbench-core-authorization';
```
[the remark this answers is not in the document]
Comment (insignias) [reply solutions/swb-app/src/apiRouteConfig.ts:7]: just remove it for now then ?

[the remark this answers is not in the document]
Comment (aws-tyler) [reply solutions/swb-app/src/apiRouteConfig.ts:7]: Yup, i have an interface ready in my branch that we can use.

File: solutions/swb-reference/src/backendAPI.ts [hunk 2]
```diff
@@ -31,6 +32,25 @@ const aws: AwsService = new AwsService({
   region: process.env.AWS_REGION!,
   ddbTableName: process.env.STACK_NAME!
 });
+
+// Dynamic Auth
+const dynamicAuthAws: AwsService = new AwsService({
```
[the remark this answers is not in the document]
Comment (insignias) [reply solutions/swb-reference/src/backendAPI.ts:37]: Another DynamoDB table is used by DynamicAuth

File: solutions/swb-reference/src/SWBStack.ts [hunk 3]
```diff
@@ -820,8 +846,42 @@ export class SWBStack extends Stack {
...
+      gsis: [
+        {
+          indexName: 'getIdentityPermissionsByIdentity',
+          partitionKey: { name: 'Identity', type: AttributeType.STRING },
+          sortKey: { name: 'pk', type: AttributeType.STRING }
+        }
+      ]
+    });
+
+    // eslint-disable-next-line no-new
```
[the remark this answers is not in the document]
Comment (insignias) [reply solutions/swb-reference/src/SWBStack.ts:871]: https://eslint.org/docs/latest/rules/no-new

File: solutions/swb-reference/src/backendAPI.ts [hunk 4]
```diff
@@ -31,6 +32,25 @@ const aws: AwsService = new AwsService({
...
+  region: process.env.AWS_REGION!,
+  ddbTableName: process.env.STACK_NAME!
+});
+
+const wbcGroupManagementPlugin: WBCGroupManagementPlugin = new WBCGroupManagementPlugin({
+  userManagementService: new UserManagementService(
+    new CognitoUserManagementPlugin(process.env.USER_POOL_ID!, aws)
+  ),
+  ddbService: dynamicAuthAws.helpers.ddb,
+  userGroupKeyType: 'GROUP'
```
[the remark this answers is not in the document]
Comment (insignias) [reply solutions/swb-reference/src/backendAPI.ts:47]: I was trying to follow the same convention as done [here](https://github.com/aws-solutions/solution-spark-on-aws/blob/develop/solutions/swb-reference/src/backendAPI.ts#L65). But sure :)

File: solutions/swb-reference/src/SWBStack.ts [hunk 5]
```diff
@@ -166,7 +171,28 @@ export class SWBStack extends Stack {
...
     );
-    this._createDDBTable(apiLambda, statusHandler, createAccountHandler);
+
+    // Create DataSet DynamoDB Table
+    this._createDataSetDDBTable(apiLambda, statusHandler, createAccountHandler);
+
+    // Encryption Key
+    const dynamodbEncryptionKey: WorkbenchEncryptionKeyWithRotation = new WorkbenchEncryptionKeyWithRotation(
+      this,
+      'DynamicAuthDynamodbEncryptionKey'
```
[the remark this answers is not in the document]
Comment (insignias) [reply solutions/swb-reference/src/SWBStack.ts:181]: this is an identifier for the resource being created

**Comment by insignias** [comment]
> To prevent merge conflicts from happening with my branch, lets remove the `swb-app` code. We're not allowing any new references to external projects in `swb-app` so we can make sure we're not tightly coupling with anything that could change without our knowledge.
> 
Sure, I can open a PR directly to you branch then ! 

> The only other thing is I noticed there are no tests. Is there anything we can do to make sure the infrastructure doesn't change unexpectedly?
> 
We can add fine grained assertions or a SNAPSHOT test. Fine grained assertion is going to take little time and I used SNAPSHOT tests for example package which everyone was annoyed about SNAPSHOTS not getting updated properly so I ended up removing it.

> Once this is updated, I can pull this into my branch ASAP and we can get the service merged into develop 🎉🎉🎉
Sounds good !

File: solutions/swb-reference/src/SWBStack.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (aws-tyler) [reply solutions/swb-reference/src/SWBStack.ts:870]: it looks like we already do this for the entire file: `/* eslint-disable no-new */`. Was the linter complaining? if so we should figure out whats going on so we can manage this in a single place.

**Review by aws-tyler — DISMISSED** [review]

File: solutions/swb-reference/src/backendAPI.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (insignias) [reply solutions/swb-reference/src/backendAPI.ts:38]: you're right ! updated, it was a copy error

**Review by aws-tyler — APPROVED** [review]

**Review by htztomic — APPROVED** [review]

**Review by htztomic — APPROVED** [review]

**Review by aws-tyler — APPROVED** [review]

**Pull request merged**